### PR TITLE
Add Cookie SameSite

### PR DESCRIPTION
**Pull request by kainonly**

### CODE REVIEW

**Comment by liu21st** [comment]
该属性用法 PHP7.3才支持的吧

**Comment by kainonly** [comment]
7.3.0 | An alternative signature supporting an options array has been added. This signature supports also setting of the SameSite cookie attribute.
-- | --
5.5.0 | A Max-Age attribute is now included in the Set-Cookie header sent to the client.
5.2.0 | The httponly parameter was added.

的确是这样，版本兼容这个之前没考虑到

File: src/think/Cookie.php [hunk 1]
```diff
@@ -197,11 +208,32 @@ public function save(): void
...
-    protected function saveCookie(string $name, string $value, int $expire, string $path, string $domain, bool $secure, bool $httponly): void
+    protected function saveCookie(
+        string $name,
+        string $value,
+        int $expire,
+        string $path,
+        string $domain,
+        bool $secure,
+        bool $httponly,
+        string $samesite
```
Comment (hongweipeng) [inline src/think/Cookie.php]: 默认配置 `samesite => null` ，这里严格模式下会报错。

Comment (hongweipeng) [reply src/think/Cookie.php]: 抱歉，我应该讲清楚一点。samesite的默认值应该是空字符比较合理。php.ini里默认的配置也是空字符串。且我测试了空串时，samesite没有发送到响应头部去（php7.3.10_x64_windows）。

**Pull request merged**